java/src/main/java/com/google/oak/client/OakClient.java

```diff
@@ -16,67 +16,95 @@
 
 package com.google.oak.client;
 
-import com.google.oak.evidence.Evidence;
+import com.google.oak.crypto.ClientEncryptor;
+import com.google.oak.crypto.Encryptor;
+import com.google.oak.remote_attestation.AttestationVerifier;
+import com.google.oak.transport.EvidenceProvider;
+import com.google.oak.transport.Transport;
 import com.google.oak.util.Result;
+import oak.session.noninteractive.v1.AttestationBundle;
+import oak.session.noninteractive.v1.AttestationEndorsement;
+import oak.session.noninteractive.v1.AttestationEvidence;
 
 /**
- * Generic client interface for sending and receiving encrypted messages to a server.
+ * Oak Client class for exchanging encrypted messages with an Oak Enclave which is being run by the
+ * Oak Launcher.
  *
- * <p>Three additional nested interfaces are provided to facilitate instantiating concrete
- * implementations of this interface: {@code EvidenceProvider}, {@code EncryptorProvider}, and
- * {@code RpcClientProvider}. Implementations of {@code OakClient} first verify the {@code
- * Evidence} provided by the {@code EvidenceProvider}. Successful verification yields a public key
- * that will be used by the {@code EncryptorProvider} to generate an {@code Encryptor}. The {@code
- * RpcClient} provided by the {@code RpcClientProvider} will be used to send and receive encrypted
- * messages.
- *
- * @param R type of the requests that this client sends
- * @param T type of the responses that this client receives
+ * @param T type of the transport used to communicate with an Oak Launcher.
  */
-public interface OakClient<R, T> extends AutoCloseable {
+public class OakClient<T extends Transport> {
+  private static final byte[] EMPTY_ASSOCIATED_DATA = new byte[0];
+
+  private final T transport;
+  private final byte[] serverEncryptionPublicKey;
+
   /**
-   * Sends a request to a remote server and receives the response.
+   * Oak Client class for exchanging encrypted messages with an Oak Enclave which is being run by
+   * the Oak Launcher.
    *
-   * @param request the request to send to the server
-   * @return the response received from the server wrapped in a {@code Result}
+   * @param E type that implements interfaces for the transport and for the evidence provider.
```

Comment:
A generic type parameter should be documented with `<E>`. https://docs.oracle.com/javase/1.5.0/docs/tooldocs/windows/javadoc.html#@param

Also, why is only one type parameter documented here? I would either remove the paramter, or document all type parameters, input parameters and return vlaue.

Reply:
Done